addons/sale_mrp/models/sale_order_line.py

```diff
@@ -119,8 +119,18 @@ def _get_incoming_outgoing_moves_filter(self):
         :return: Dictionary with incoming moves and outgoing moves
         :rtype: dict
         """
+        # The first move created was the one created from the intial rule that started it all.
+        sorted_moves = self.move_ids.sorted('id')
+        triggering_rule_ids = []
+        seen_wh_ids = set()
+        for move in sorted_moves:
+            if move.warehouse_id.id not in seen_wh_ids:
+                triggering_rule_ids.append(move.rule_id.id)
+                seen_wh_ids.add(move.warehouse_id.id)
+
         return {
-            'incoming_moves': lambda m: m.location_dest_id.usage == 'customer' and
+            'incoming_moves': lambda m: (m.location_dest_id.usage == 'customer' or
+                        (m.rule_id.id in triggering_rule_ids and m.location_final_id.usage == 'customer')) and
```

Comment:
`_get_incoming_outgoing_moves_filter()` is only used in `_get_qty_procurement()` 

In sale_stock,  `_get_qty_procurement()` will call `_get_outgoing_incoming_moves()` with parameter `strict=False`. Knowing that, you could rewrite the lambda as 
```python
 lambda m: (
    m.rule_id.id in triggering_rule_ids 
    and m.location_final_id.usage == 'customer'
    and (not m.origin_returned_move_id or (m.origin_returned_move_id and m.to_refund)
)
```

Reply:
@Whenrow I've also added m.state != 'cancel' and not m.scrapped